docs/design.md

```diff
@@ -1,8 +1,65 @@
-# Ansible Playbook Bundle (APB) Spec File
+# Design
+
+## Overview
+An [Ansible Playbook Bundle (APB)](https://github.com/fusor/ansible-playbook-bundle)
+borrows several concepts from the [Nulecule](https://github.com/projectatomic/nulecule)
+or [Atomicapp](http://www.projectatomic.io/docs/atomicapp/) project, namely the concept of a short
+lived container with the sole purpose of orchestrating the deployment of the intended application. For the case
+of APB, this short lived container is the APB; a container with an Ansible runtime environment
+plus any files required to assist in orchestration such as playbooks, roles, and extra dependencies.
+Specification of an APB is intended to be lightweight, consisting of several named playbooks and a
+metadata file to capture information such as parameters to pass into the application.
+
+## Workflow
+APB is broken up into the following steps.
+
+  1. [Preparation](#preparation)
+     * [APB init](#apb-initialization)
+     * [Spec File](#spec-file)
+     * [Actions](#actions) (provision, deprovision, bind, unbind)
+  1. [Build](#build)
```

Comment:
done on purpose, for any future modifications.  The numbers will automatically show up correctly, so if anything were to be inserted above the current #2, the rest of the numbers don't need to be modified..